src/main/java/bio/terra/cda/app/util/QueryTranslator.java

```diff
@@ -28,44 +26,46 @@ public String sql() {
     return String.format("SELECT * FROM %s WHERE %s", fromClause, condition);
   }
 
-  private Stream<String> getUnnestColumns(Query query) {
-    if (query.getNodeType() == Query.NodeTypeEnum.QUOTED) {
-      return Stream.empty();
-    }
-
-    if (query.getNodeType() == Query.NodeTypeEnum.UNQUOTED) {
-      return Stream.empty();
-    }
-
-    if (query.getNodeType() == Query.NodeTypeEnum.COLUMN) {
-      return Stream.of(getColumnParent(query.getValue()));
+  private static Stream<String> getUnnestColumns(Query query) {
+    switch (query.getNodeType()) {
+      case QUOTED:
+      case UNQUOTED:
+        return Stream.empty();
+      case COLUMN:
+        if (hasColumnParent(query)) {
+          return Stream.of(getColumnParent(query));
+        }
+        return Stream.empty();
+      default:
+        return Stream.concat(getUnnestColumns(query.getL()), getUnnestColumns(query.getR()));
     }
+  }
 
-    return Stream.concat(getUnnestColumns(query.getL()), getUnnestColumns(query.getR()));
+  public static boolean hasColumnParent(Query query) {
+    return query.getNodeType() == Query.NodeTypeEnum.COLUMN && query.getValue().indexOf('.') != -1;
   }
 
-  public static String getColumnParent(String column) {
-    var parts = column.split("\\.");
+  public static String getColumnParent(Query query) {
+    var parts = query.getValue().split("\\.");
     if (parts.length > 1) {
       return parts[0];
     }
     return null;
```

Comment:
I was able to remove `.filter(Objects::nonNull)` above since this line should never be executed. When we add error handling, there could be a throw here.